### PR TITLE
Fix fake data

### LINKED ISSUE #693: Fix upload parsing of site, funding type, reporting periods
Collab with kyle on #540

### PR DESCRIPTION
**Pull request by mmazanec22**
Closes #693 .  Addresses upload parsing of site and funding type, leaves reporting period formatting for #540 .  Breaks up the map file into a mapUtils folder because I was having a hard time keeping track of what was where.

When I download an example file and upload it, there are still a few funding types that aren't parsed.  I can't really make any sense of it, except maybe that the combination of funding type/time/whatever is not supported by that org?  Any help on that would be appreciated, though it is probably not essential for launch if it's just a problem with the fake data quality.

### CODE REVIEW

File: src/controllers/export.ts [hunk 1]
```diff
@@ -121,28 +121,34 @@ export async function streamUploadedChildren(
...
   }
-  if (typeof value == 'string') {
+  if (typeof value === 'string') {
     return value || '';
   }
-  if (typeof value == 'number') {
+  if (typeof value === 'number') {
     return value.toString();
   }
+  if (propertyName === 'site') {
```
Comment (mmazanec22) [inline src/controllers/export.ts:143]: This was the actual problem with exporting site data

File: src/controllers/export.ts [hunk 2]
```diff
@@ -121,28 +121,34 @@ export async function streamUploadedChildren(
  * in cells of the CSV to export.
  * @param value
  */
-function formatStringPush(value: any) {
+function formatStringPush(value: any, propertyName: string) {
```
Comment (jlhogan) [inline src/controllers/export.ts:124]: nit: this function name doesn't mean anything to me

File: src/controllers/export.ts (hunk 1 above)
Comment (jlhogan) [reply src/controllers/export.ts:143]: nit: could we put all these specific property name chekcs together (i.e. pull the funding space check down here?)

**Comment by jlhogan** [comment]
>When I download an example file and upload it, there are still a few funding types that aren't parsed. I can't really make any sense of it, except maybe that the combination of funding type/time/whatever is not supported by that org? Any help on that would be appreciated, though it is probably not essential for launch if it's just a problem with the fake data quality.

can you provide an example ? Should be easy enough to check if the failure is b/c there is no matching funding space (e.g. is not supported by the org)

**Review by jlhogan — APPROVED** [review]

**Comment by RyanHansz** [comment]
@mmazanec22 I don't believe this is a problem with fake data. I'm having batch upload issues with data I enter myself that should be valid.
Example:
[SCenariotesting.xlsx](https://github.com/ctoec/data-collection/files/5519479/SCenariotesting.xlsx)

**Comment by mmazanec22** [comment]
Ok so the anomalous ones were all cases where:
- the child had only one enrollment
- the funding time was AdditionalExtendedSchool

**Comment by RyanHansz** [comment]
> Ok so the anomalous ones were all cases where:
> 
> * the child had only one enrollment
> * the funding time was AdditionalExtendedSchool

Added issue with SHS funding spaces here: 
https://github.com/ctoec/data-collection/issues/718

File: src/controllers/export.ts [hunk 3]
```diff
@@ -121,28 +121,35 @@ export async function streamUploadedChildren(
...
-    return value ? 'Yes' : 'No';
+  if (
+    propertyName === 'firstReportingPeriod' ||
+    propertyName === 'lastReportingPeriod'
+  ) {
+    return reportingPeriodToString(value);
   }
-  if (isMoment(value)) {
-    return value.format('MM/DD/YYYY');
+  // Name matching means some values are nested one more level
```
Comment (bamader) [inline src/controllers/export.ts:140]: Nit: can delete this since you have the comment about checking for specific names

**Comment by mmazanec22** [comment]
Ok I figured out the issue for #718 -- the funding space table was only allocating 20 characters to funding times.  Migration is included in this PR.

**Review by bamader — APPROVED** [review]

**Comment by jlhogan** [comment]
>Ok I figured out the issue for #718 -- the funding space table was only allocating 20 characters to funding times. Migration is included in this PR.

I have a change in an open PR that changes the way we're doing enums to not use varchar with arbitrary length and actually enforce enum value check at DB level so things like this should never happen again :)

**Pull request merged**